.github/workflows/build_and_test.yml

```diff
@@ -217,6 +217,16 @@ jobs:
       run: |
         python3.6 -m pip install numpy 'pyarrow<3.0.0' pandas scipy xmlrunner plotly>=4.8
         python3.6 -m pip list
+    # TODO(SPARK-35507) Move Python 3.9 installtation to the docker image
+    - name: Install Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+        architecture: x64
+    - name: Install Python packages (Python 3.9)
+      run: |
+        python3.9 -m pip install numpy 'pyarrow<5.0.0' pandas scipy xmlrunner plotly>=4.8
```

Comment:
Is this intentional to add a new PyArrow version test coverage on Python 3.9 only?

Reply:
Oh yeah. I should've commented here. Python 3.9 support was added from https://issues.apache.org/jira/browse/ARROW-10224, and I just tentatively tried PyArrow 4.0.0 but it worked. So I just set it to the highest working version for now.

Reply:
cc @BryanCutler FYI

Reply:
SGTM. The Arrow binary format remains the same so it's good to continue testing with the latest pyarrow.